packages/container/lib/container.js

```diff
@@ -421,15 +421,16 @@ function deprecatedFactoryFor(container, fullName, options = {}) {
   }
 
   let cache = container.factoryCache;
-  if (cache[fullName]) {
-    return cache[fullName];
+  let cached = cache[fullName];
+  if (cached) {
+    return cached;
   }
   let factory = registry.resolve(fullName);
   if (factory === undefined) {
     return;
   }
 
-  let type = fullName.split(':')[0];
+  let [type] = fullName.split(':');
```

Comment:
yes, but it must run that code. If we leave it without destructuring it is not required.

object descturing `{ foo} = bar` works fine, but array destructuring is abit dubious unless the array is in scope`[first] = [1,2,3]`